content/exsaml.xqm

```diff
@@ -373,13 +378,18 @@ declare %private function exsaml:validate-saml-assertion($assertion as item()) {
 };
 
 (: retrieve issued SAML request id and delete if answered :)
+(: this is pointless if we are configured to accept unsolicited SAML assertions :)
 declare %private function exsaml:check-authnreqid($reqid as xs:string) {
-    let $log := exsaml:log("info", "verifying SAML request id: " || $reqid)
-    return
-        if (system:as-user($exsaml:exsaml-user, $exsaml:exsaml-pass,
-                exists(doc($exsaml:saml-coll-reqid||"/"||$reqid)) and empty(xmldb:remove($exsaml:saml-coll-reqid, $reqid)))) then
-            $reqid
-        else ""
+    if (not($exsaml:idp-unsolicited)) then (
```

Comment:
Same as above, I think it is better readable if the logic is inverted here to read:
if unsolicited requests are acceptable do nothing
else do this ...

Reply:
ok, and comment added for clarity